In [1]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt

**Importing Data**

In [2]:
df = pd.read_csv('Diabities-210331-154610.csv')

In [3]:
# Snapshot of the dataframe

df.head()

Pregnancies  Glucose  blood pressure  skin thickness  Insulin   BMI  \
0            6      148              72              35        0  33.6   
1            1       85              66              29        0  26.6   
2            8      183              64               0        0  23.3   
3            1       89              66              23       94  28.1   
4            0      137              40              35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   blood pressure            768 non-null    int64  
 3   skin thickness            768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [5]:
df.isnull().values.any()

False

In [6]:
# List the columns

df.columns

Index(['Pregnancies', 'Glucose', 'blood pressure', 'skin thickness', 'Insulin',
       'BMI', 'DiabetesPedigreeFunction', 'Age', 'Outcome'],
      dtype='object')

In [7]:
X = df.iloc[:, :-1].values
y = df.iloc[:, -1].values

In [8]:
print(X.shape)

(768, 8)


In [9]:
print(y.shape)

(768,)


**Data Split**

In [10]:
from sklearn.model_selection import train_test_split

In [11]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

In [12]:
num_rows_train = len(y_train)
print('The number of entries in the train split are:', num_rows_train)

num_rows_test = len(y_test)
print('The number of entries in the test split are:', num_rows_test)

The number of entries in the train split are: 614
The number of entries in the test split are: 154


**Train the model(s)**

## RandomForestClassifier

In [13]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score

In [14]:
clf = RandomForestClassifier()
clf.fit(X_train, y_train)
y_pred = clf.predict(X_test)

In [17]:
acc_random_forest = round(accuracy_score(y_pred, y_test) * 100, 2)
print("Accuracy:", acc_random_forest, "%")

Accuracy: 79.22 %


## LogisticRegression

In [18]:
from sklearn.linear_model import LogisticRegression

In [20]:
clf = LogisticRegression(solver = 'lbfgs', max_iter = 1000)
clf.fit(X_train, y_train)
y_pred = clf.predict(X_test)

In [21]:
acc_log_reg = round(accuracy_score(y_pred, y_test) * 100, 2)
print("Accuracy:", acc_log_reg, "%")

Accuracy: 82.47 %


## KNeighbour Classifier

In [22]:
from sklearn.neighbors import KNeighborsClassifier

In [23]:
knn = KNeighborsClassifier()
knn.fit(X_train, y_train)
y_pred = knn.predict(X_test)

In [24]:
acc_knn = round(accuracy_score(y_pred, y_test) * 100, 2)
print("Accuracy:", acc_knn, "%")

Accuracy: 75.32 %


As we can see **Logistic Regression** model performs the best with an accuracy score of `82.47%`.